auth/src/lib.rs

```diff
@@ -54,37 +58,38 @@ pub async fn sync(pool: PgPool, args: SyncArgs) -> Result<()> {
         match permit_client.get_user(&user.id).await {
             Ok(p_user) => {
                 // Update tier if out of sync
+                let wanted_tier = user.account_tier.as_permit_account_tier();
                 if !p_user
                     .roles
-                    .is_some_and(|rs| rs.iter().any(|r| r.role == user.account_tier.to_string()))
+                    .is_some_and(|rs| rs.iter().any(|r| r.role == wanted_tier.to_string()))
                 {
-                    match user.account_tier {
-                        AccountTier::Basic
-                        | AccountTier::PendingPaymentPro
-                        | AccountTier::CancelledPro
-                        | AccountTier::Team
-                        | AccountTier::Admin
-                        | AccountTier::Deployer => {
+                    println!("updating tier for user: {}", user.id);
+                    match wanted_tier {
+                        AccountTier::Basic => {
                             permit_client.make_basic(&user.id).await?;
                         }
                         AccountTier::Pro => {
                             permit_client.make_pro(&user.id).await?;
                         }
+                        _ => unreachable!(),
                     }
                 }
             }
-            Err(_) => {
-                // FIXME: Make the error type better so that this is only done on 404s
-
+            Err(Error::ResponseError(ResponseContent {
+                status: StatusCode::NOT_FOUND,
+                ..
+            })) => {
+                // Add users that are not in permit
                 println!("creating user: {}", user.id);
 
-                // Add users that are not in permit
                 permit_client.new_user(&user.id).await?;
-
-                if user.account_tier == AccountTier::Pro {
+                if user.account_tier.as_permit_account_tier() == AccountTier::Pro {
```

Comment:
why is this only for pro?

Reply:
Only basic and pro tiers exist in Permit. (other ones don't affect permissions)